app/src/main/java/protect/card_locker/MainActivity.java

```diff
@@ -413,8 +413,12 @@ protected void onResume() {
             addButton.setOnClickListener(v -> {
                 Intent intent = new Intent(getApplicationContext(), ScanActivity.class);
                 Bundle bundle = new Bundle();
-                if (selectedTab != 0) {
-                    bundle.putString(LoyaltyCardEditActivity.BUNDLE_ADDGROUP, groupsTabLayout.getTabAt(selectedTab).getText().toString());
+                TabLayout.Tab tab = groupsTabLayout.getTabAt(selectedTab);
+                if (tab != null) {
+                    Group group = (Group) groupsTabLayout.getTabAt(selectedTab).getTag();
```

Comment:
Am I missing something here or was this unnecessary duplication?

```suggestion
                    Group group = (Group) tab.getTag();
```